nulib/dc-nextjs
# Move Gen AI toggle tab state to localStorage

## What does this do?
- Defaults the active tab on Search page to show Chat streaming response when a user enters the Search route with the Gen AI checkbox checked.
- Moves Gen AI checkbox state from URL query params, to `localStorage`
- Removes active tab state from Search Context
- Attempts to handle faceting/filtering UX when Gen AI checkbox state is active

This PR moves the ball down the field, however there is one last piece which I'm not sure of the proper direction?

When a user is search w/ Gen AI, switches to the View Results tab, actively filters the results, and then clicks on the "AI Response" tab, the AI stream creates a new websocket request and doesn't render the cached result.   Is this an expected outcome or no?  Does faceting or filtering items change the nature of the original AI response, I guess is the question.

## How to test
Basically just click around the app, searching, toggling the Gen AI check box on off at various places and times.  Try logging in / out, see how the UX responds.  Anything which seems out of the ordinary should be noted and addressed.
